File: ios/ExpoSquircleViewModule.swift
import ExpoModulesCore

public class ExpoSquircleViewModule: Module {
    public func definition() -> ModuleDefinition {
        Name("ExpoSquircleView")
        
        View(ExpoSquircleView.self) { 
            Prop("squircleBackgroundColor") { (view: ExpoSquircleView, prop: UIColor) in
                view.setBackgroundColor(prop)
            }
            Prop("squircleBorderColor") { (view: ExpoSquircleView, prop: UIColor) in
                view.setBorderColor(prop)
            }
            Prop("squircleBorderWidth") { (view: ExpoSquircleView, prop: Float) in
                view.setBorderWidth(CGFloat(prop))
            }
            Prop("borderRadius") { (view: ExpoSquircleView, prop: Float) in
                view.setRadius(CGFloat(prop))
            }
            Prop("cornerSmoothing") { (view: ExpoSquircleView, prop: Int) in
                view.setCornerSmoothing(CGFloat(prop))
            }
            Prop("preserveSmoothing") { (view: ExpoSquircleView, prop: Bool) in
                view.setPreserveSmoothing(prop)
            }
            Prop("enabledIOSAnimation") { (view: ExpoSquircleView, prop: Bool) in
                view.isAnimationEnabled = prop
            }
        }
    }
}

File: android/src/main/java/expo/modules/squircleview/ExpoSquircleViewModule.kt
package expo.modules.squircleview

import expo.modules.kotlin.modules.Module
import expo.modules.kotlin.modules.ModuleDefinition

class ExpoSquircleViewModule : Module() {
    override fun definition() = ModuleDefinition {

        Name("ExpoSquircleView")

        View(ExpoSquircleView::class) {
            Prop("squircleBackgroundColor") { view: ExpoSquircleView, prop: Int? ->
                if (prop != null) {
                    view.setViewBackgroundColor(prop)
                }
            }

            Prop("squircleBorderColor") { view: ExpoSquircleView, prop: Int? ->
                if (prop != null) {
                    view.setBorderColor(prop)
                }
            }

            Prop("squircleBorderWidth") { view: ExpoSquircleView, prop: Float ->
                view.setBorderWidth(prop)
            }

            Prop("borderRadius") { view: ExpoSquircleView, prop: Float ->
                view.setBorderRadius(prop)
            }

            Prop("cornerSmoothing") { view: ExpoSquircleView, prop: Int ->
                view.setCornerSmoothing(prop)
            }

            Prop("preserveSmoothing") { view: ExpoSquircleView, prop: Boolean ->
                view.setPreserveSmoothing(prop)
            }
        }
    }
}
